quinn-proto/src/connection/mod.rs

```diff
@@ -5568,7 +5605,7 @@ impl Connection {
         let ack_delay_exp = TransportParameters::default().ack_delay_exponent;
         let delay = delay_micros >> ack_delay_exp.into_inner();
 
-        if send_path_acks {
+        if is_multipath_negotiated && space_id == SpaceId::Data {
```

Comment:
Yes, this is correct I think. The bug was because on the client side multipath was already negotiated (the handshake completed!) but it is still sending ACKs in the Initial and Handshake space after this.

The choice of moving that decision here by plugging SpaceId & is_multipath_negotiated through is fine I guess.